src/renderer/containers/App/index.tsx

```diff
@@ -115,11 +117,17 @@ export default function App() {
       // the addition of the fileId i.e. FSS's completion, has failed,
       // or requires this clients intervention to retry
       if (
-        job.service === Service.FILE_STORAGE_SERVICE &&
-        (job.serviceFields?.fileId || job.status === JSSJobStatus.FAILED || job.currentStage === UploadStatus.RETRY)
-      ) {
-        dispatch(receiveFSSJobCompletionUpdate(job as FSSUpload));
-      } else if (job.serviceFields?.type === "upload") {
+        job.service === Service.FILE_STORAGE_SERVICE) {
+          if (job.serviceFields?.fileId || job.status === JSSJobStatus.FAILED || job.currentStage === UploadStatus.RETRY) {
+            dispatch(receiveFSSJobCompletionUpdate(job as FSSUpload));
+          } else if(job.serviceFields?.preUploadMd5 && (job.serviceFields.preUploadMd5 !== job.serviceFields?.fileSize)) {
```

Comment:
Are `job.serviceFields.preUploadMd5` and `job.serviceFields.postUploadMd5` file sizes not MD5 hashes?

Reply:
The are the number of bytes that have been read so far while computing the md5 hashes (the progress).

Reply:
Btw, do you have ideas on how to unit this this code?  verify that the actions are dispatched?  I havent seen an example in this codebase.